azurerm/internal/services/keyvault/resourceids.go

```diff
@@ -1,3 +1,4 @@
 package keyvault
 
 //go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=Vault -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.KeyVault/vaults/vault1
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=HardwareSecurityModule -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.KeyVault/managedHSMs/hsm1
```

Comment:
this is a Managed hardware security module, so presumably this can become:

```suggestion
//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=ManagedHSM -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.KeyVault/managedHSMs/hsm1
```

which also fixes the naming to be `Name` rather than `H S M`